web/src/NoteView/BarRenderer.ts

```diff
@@ -57,47 +59,70 @@ export class BarRenderer extends Drawable<BarRendererProps> {
     const { beat, key } = guideDotOnMouse;
     const beatKey = { beat, key };
 
-    dispatch(BarAction.clickBar(new Position({
-      barId: this.props.bar.id,
-      beat,
-      key,
-    })));
-
-    if (mode === 'singleNoteEdit') {
-      switch (event.button) {
-        case 0: { // left click
-          const noteOnBeatKey = this.getNoteOnBeatKey(beatKey)
-          if (noteOnBeatKey) {
-            if (noteOnBeatKey === this.props.selectedNote) {
-              this.deselectNote()
-            } else {
-              this.selectNote(noteOnBeatKey);
-            }
+    switch (event.button) {
+      case 0: { // left click
+        dispatch(BarAction.clickBar(new Position({
+          barId: this.props.bar.id,
+          beat,
+          key,
+        })));
+
+        if (mode === 'longNoteEdit') {
+          break;
+        }
+
+        const noteOnBeatKey = this.getNoteOnBeatKey(beatKey);
+        if (noteOnBeatKey) {
+          if (noteOnBeatKey === this.props.selectedNote) {
+            this.deselectNote()
           } else {
-            this.addNote(beatKey);
-          }
-        } break;
-
-        case 2: { // right click
-          const noteOnBeatKey = this.getNoteOnBeatKey(beatKey)
-          if (noteOnBeatKey) {
-            this.removeNote(noteOnBeatKey);
+            this.selectNote(noteOnBeatKey);
           }
-        } break;
-      }
+        } else {
+          this.addNote(beatKey);
+        }
+      } break;
+
+      case 2: { // right click
+        if (mode === 'longNoteEdit') {
+          ModeSelectHandler.cancelAddLongNoteProcess();
```

Comment:
May we break switch here..?

Reply:
This makes deleting note possible even in long note edit mode.